src/components/EditTaskCard/EditTaskCard.jsx

```diff
@@ -0,0 +1,82 @@
+import { useState } from "react";
+import { useDispatch } from "react-redux";
+import PropTypes from "prop-types";
+import { editTask, markTaskDone } from "src/store/task/actions/taskActions";
+import {
+  ALT_TEXT_DELETE_ICON,
+  ALT_TEXT_DONE_ICON,
+  ICON_DELETE,
+  ICON_DONE,
+} from "src/common/constants";
+import { sanitizeText } from "src/utils/sanitizeText";
+import IconButton from "src/components/Buttons/IconButton/IconButton";
+import "src/components/AddTaskCard/AddTaskCard.scss";
+
+function EditTaskCard({ taskId, taskTitle, onToggleEditTask }) {
+  const [inputText, setInputText] = useState(taskTitle);
+  const sanitizedInput = sanitizeText(inputText);
+  const isSaveButtonDisabled = sanitizeText(inputText) === "";
+  const dispatch = useDispatch();
+
+  function handleChange(event) {
+    setInputText(event.target.value);
+  }
+
+  function handleSave() {
+    if (sanitizedInput !== taskTitle) {
+      dispatch(editTask({ taskId, taskTitle: sanitizedInput }));
+    }
+
+    onToggleEditTask();
+  }
+
+  function handleDelete() {
+    setInputText(taskTitle);
+  }
+
+  function handleDone() {
+    dispatch(markTaskDone(taskId));
+    handleSave();
+  }
+
+  return (
+    <>
+      <textarea
+        className="task__textarea"
+        autoFocus
+        value={inputText}
+        onChange={handleChange}
+      />
+
+      <div className="tasks-container__box-buttons">
+        <button
+          className={`task__save-button ${
+            isSaveButtonDisabled ? "task__save-button-disabled " : null
+          }`}
+          disabled={isSaveButtonDisabled}
+          onClick={handleSave}
+        >
```

Comment:
Use component instead of button tag.

Reply:
@rsa-wtag I want to keep the icon buttons (which needs some sort img file as there source) and normal text buttons separate. 
That's why not preferring to create another component for that.

Reply:
@nsi-wtag why are we using two separate ways to render buttons, when we can reuse button component for both cases?

Reply:
@rsa-wtag we are using two separate kinds of buttons because of the differences between their content. The `IconButton` component has some sort of image file as its content. Whereas, other button (for which we are using the html `button` tag) has text as its content. 

So, in this case I am guessing the separation is a cleaner approach and more to the point. What do you think?